DOCS-10905: Anypoint DataGraph commands (#491)
mulesoft/docs-runtime-manager: modules/ROOT/pages/anypoint-platform-cli-commands.adoc

> account business-group list [options]
----

> This command displays all xref:access-management::organization.adoc#business-groups[business groups]. It returns the name of the business group, the type ('Master' or 'Business unit') and the Id.

Reviewer: ```suggestion
This command displays all xref:access-management::organization.adoc#business-groups[business groups]. It returns the name of the business group, the type ('Master' or 'Business unit'), and the ID.
```

Reviewer: See style guide on "ID": https://confluence.internal.salesforce.com/display/MTDT/Term+Presentation+Reference#TermPresentationReference-I

Author: Yeah, I know. I didn't edit that content - just replaced the boilerplate with the include, which apparently triggered a whole lot of green.

|`--sort` | Sorts the results in the field name passed | `api-mgr tier list --sort "Latest Version"`
|===

=== datagraph load-balancer-config add

----
> datagraph load-balancer-config add <dlbUrl>
----

> This command adds a dedicated load balancer configuration specified by `<dlbUrl>` to Anypoint DataGraph.

Reviewer: Does dlbUrl mean the URL of the dedicated load balancer? Description says "a dedicated load balancer configuration," which sounds more general. Maybe people can add things other than a URL?

Author: No. That's what the CLI source says.

Reviewer: I wonder if the descriptions in the source are mixed up bc of what I see below/next.

> This command adds a dedicated load balancer configuration specified by `<dlbUrl>` to Anypoint DataGraph.

Author: ```suggestion
This command adds a dedicated load balancer configuration specified by `<dlbUrl>` to Anypoint DataGraph.
The `dlbUrl` is a valid URL that includes the DLB domain and the mapping rule `inputUri`.

```

// CLI DEFAULTS
include::partial$cli-default-options.adoc[tag=CLIdefaultOptions]

=== datagraph load-balancer-config describe

----
> datagraph load-balancer-config describe [options]
----

> This command displays a dedicated load balancer URL for Anypoint DataGraph.

Reviewer: this one talk about the dlbURL. Was there possibly a mixup with the previous?

[WARNING]
This command does not prompt twice before deleting. If you send a delete instruction, it does not ask for confirmation.

// CLI DEFAULTS
include::partial$cli-default-options.adoc[tag=CLIdefaultOptions]

=== designcenter project download

----
> designcenter project download [options] <name> <targetDir>
----

> This Command downloads the Design Center project passed in `name` to your local directory specified in `targetDir`.

Reviewer: ```suggestion
This command downloads the Design Center project passed in `name` to your local directory specified in `targetDir`.
```

This command deprecates the asset passed in `<assetIdentifier>`.

Argument `assetIdentifier` should be formatted as follows: `([group_id]/)<asset_id>/<version>`. +

> If `group_id` is not specified, it defaults to the currently selected Organization ID. 

Reviewer: Guess it's obvious enough here (and below) that "it" means the value of the group ID setting.

Author: That's the way Kim prefers it - that you don't repeat the subject.

----
This command describes the history of the alarm passed in `<name>`.

// CLI DEFAULTS
include::partial$cli-default-options.adoc[tag=CLIdefaultOptions]

=== runtime-mgr cloudhub-application list

----
> runtime-mgr cloudhub-application list [options]
----

> This command lists all applications available in your Anypoint Platform CLI. It returns your application name, its status, the number of vCores assigned and the last time it was updated. 

Reviewer: do we use Oxford comma here? "... assigned, and the..."

Author: Yes, but I'm not editing that text right now.

This command displays information on the application you pass in `<name>`. +

> You can start typing your application's name and press `tab` for Anypoint Platform CLI to autocomplete it, or you can double tap `tab` for a full list of all the values you can pass. +

Reviewer: Is "double tab" hyphenated? "double-tap"?

Author: Yes, but I'm leaving rewriting until next time.

> It will return data such as the application's domain, its status, last time it was updated, the Mule version, the ZIP file name, the region, monitoring, and workers; as well as `TRUE` or `FALSE` information for persistent queues and static IPs enablement. 

Reviewer: I'm not sure about "data such as ... monitoring, and workers; as well as ...".

Author: Yep!

> It will return data such as the application's domain, its status, last time it was updated, the Mule version, the ZIP file name, the region, monitoring, and workers; as well as `TRUE` or `FALSE` information for persistent queues and static IPs enablement. 

Reviewer: ```suggestion
The command returns data such as the application's domain, its status, the last time it was updated, the Mule version, the ZIP file name, the region, monitoring, and workers; as well as `TRUE` or `FALSE` information for persistent queues and static IPs enablement. 
```

// CLI DEFAULTS
include::partial$cli-default-options.adoc[tag=CLIdefaultOptions]

=== runtime-mgr cloudhub-application describe-json

----
> runtime-mgr cloudhub-application describe-json  [options] <name>
----

This command returns the raw JSON response of the application you specify in `<name>`. 

> You can start typing your application's name and press `tab` for Anypoint Platform CLI to autocomplete it, or you can double tap `tab` for a full list of all the values you can pass. 

Reviewer: same Q re "double-tap" and all instances below. 

Also, could this be an include?

Author: Not updating everything just now.

> You can start typing your application's name and press `tab` for Anypoint Platform CLI to autocomplete it, or you can double tap `tab` for a full list of all the values you can pass. 

Reviewer: ```suggestion
You can start typing your application's name and press `tab` for Anypoint Platform CLI to autocomplete it, or you can double-tap `tab` for a full list of all the values you can pass. 
```

> runtime-mgr cloudhub-application copy [options] <source> <target>
----

This command copies the CloudHub application passed in `source` to the target passed in `target`. 

> Arguments `source` and `target` should be formatted as follows: `([group_id]/)<asset_id>/<version>`. 

Reviewer: Is "should be" verboten?

Author: Yep!

> runtime-mgr standalone-application artifact [options] <identifier> <directory>
----

> This command downloads the application artifact of the `identifier` application, to the directory passed in `directory`. 

Reviewer: Is the comma needed "application, to"?

Author: Nope!

> runtime-mgr standalone-application describe-json [options] <identifier>
----

> This command describes the standalone (on-premises) application passed in `identifier` as a raw JSON response.

Reviewer: What do we mean by a standalone application? We have standalone instances of Mule. Do they mean applications deployed on-premises?

Author: Yep!

> runtime-mgr standalone-application start [options] <identifier>
----

> This command starts the standalone (on-premises) application passed in `identifier`. 

Reviewer: Same Q re standalone app. Same goes for remaining instances of this.

> runtime-mgr standalone-application restart [options] <identifier>
----

This command restarts the standalone (on-premises) application passed in `identifier`. 

// CLI DEFAULTS
include::partial$cli-default-options.adoc[tag=CLIdefaultOptions]

=== runtime-mgr standalone-application stop

----
> runtime-mgr standalone-application stop [options] <identifier>
----

This command stops the standalone (on-premises) application passed in `identifier`.

// CLI DEFAULTS
include::partial$cli-default-options.adoc[tag=CLIdefaultOptions]

=== runtime-mgr standalone-application copy

----
> runtime-mgr standalone-application copy [options] <source> <target> <targetIdentifier>
----

> This command copies the standalone (on-premises) application passed in `source` to the target passed in `target` and the server, server group or cluster ID or Name passed in `targetIdentifier`. 

Reviewer: Should it be "... server group or cluster ID, or Name passed ..."?

Author: Nope - it should be "name"

Both arguments `source` and `destination` are represented using the format: `<organizationName>:<environmentName>/<appName>`, for example:

----
> runtime-mgr standalone-application copy Services:QA/application-1 Development:QA/application-2 123456
----

> Copies the application named `application-1` from the QA environment of the _Services_ organization to the QA environment of the `_Development_` organization in the server Id 123456. 

Reviewer: ```suggestion
Copies the application named `application-1` from the QA environment of the _Services_ organization to the QA environment of the `_Development_` organization in the server ID 123456. 
```

> runtime-mgr server token [options]
----

> This command gets server registration token. This token needs to be used to register a new server. 

Reviewer: "gets a registration token" or "the registration token"

Author: Yep!

> runtime-mgr serverGroup create [options] <name> [serverIds...]
----

> This command creates a server group with the name passed in `name` using the server Id(s) passed as argument(s) thereafter.

Reviewer: "using all IDs"? do we allow constructions like "argument(s)"?

Reviewer: "arguments"

Author: Ditto

> runtime-mgr serverGroup list [options]
----

This command lists all server groups in the environment. 

// CLI DEFAULTS
include::partial$cli-default-options.adoc[tag=CLIdefaultOptions]

=== runtime-mgr serverGroup remove server

----
> runtime-mgr serverGroup remove server [options] <serverGroupId> <serverId>
----

> This command removes the removes the server passed in `serverId`, from the server group passed in `serverGroupId`. 

Reviewer: ```suggestion
This command removes the server passed in `serverId` from the server group passed in `serverGroupId`. 
```

> runtime-mgr standalone-alert list [options]
----

> This command lists all alerts for standalone Mules in the current environment. 

Reviewer: I don't think we allow "Mules." Prob should be "Mule instances".

Author: Yep!

----
> cloudhub load-balancer list [options]
----

> This command lists all load balancers in your Anypoint Platform. It displays load balancer's name, domain, its state and the Anypoint VPC Id to which the load balancer is bound. 

Reviewer: ```suggestion
This command lists all load balancers in your Anypoint Platform. It displays load balancer's name, domain, its state, and the Anypoint VPC ID to which the load balancer is bound. 
```

// CLI DEFAULTS
include::partial$cli-default-options.adoc[tag=CLIdefaultOptions]

=== cloudhub load-balancer describe

----
> cloudhub load-balancer describe [options] <name>
----
This command displays information about the load balancer that is specified in `<name>`. 

You can start typing your load balancer's name and press`tab` for Anypoint Platform CLI to autocomplete it, or you can double tap `tab` for a full list of all the values you can pass.

It displays load balancer's name, domain, its state and the Anypoint VPC Id to which the load balancer is bound.

// CLI DEFAULTS
include::partial$cli-default-options.adoc[tag=CLIdefaultOptions]

=== cloudhub load-balancer describe-json

----
> cloudhub load-balancer describe [options] <name>
----

> This command displays raw JSON response of the load balancer that is specified in `<name>`. 

Reviewer: ```suggestion
This command displays a raw JSON response from the load balancer that is specified in `<name>`. 
```

----
> cloudhub region list [options]
----
This command lists all supported regions. 

// CLI DEFAULTS
include::partial$cli-default-options.adoc[tag=CLIdefaultOptions]

=== cloudhub runtime list
----
> cloudhub runtime list [options]
----

> This command lists all supported runtimes.

Reviewer: Are we talking about Mule runtime engines? Or some other runtime. Only use "runtimes" alone if Mule is understood from the context.

----
> cloudhub vpc business-groups add [options] <vpc> <businessGroups...>
----

> This command assigns the Anypoint VPC defined in `<vpc>` to the business group(s) passed as argument(s) thereafter.

Reviewer: Do we allow constructions like "group(s)" and "argument(s)" ?

Reviewer: "passed thereafter" works for me. Not sure if we normally use it.

Author: Nope to the first.

----
> use environment [options] <name>
----

> This command makes active the environment specified in `<name>`. 

Reviewer: "makes active" -> "activates"?
